test/2fa.test.js

```diff
@@ -0,0 +1,251 @@
+            .expect((res) => {
+                assert.equal(res.body.success, true);
+                securityCode = getCodeFromLogs();
+            })
+            .expect(200);
+    });
+
+    test('should be able to call init again with different method', async () => {
+
+        method = twoFactorMethods[1];
+
+        await request.post('/2fa/init')
+            .send({
+                accountId,
+                method,
+                ...(await signatureFor(accountId))
+            })
+            .expect('Content-Type', /json/)
+            .expect((res) => {
+                assert.equal(res.body.success, true);
+                securityCode = getCodeFromLogs();
+            })
+            .expect(200);
+    });
+
+    test('verify 2fa method', async () => {
+
+        await request.post('/2fa/verify')
+            .send({
+                accountId,
+                requestId,
+                securityCode,
+                ...(await signatureFor(accountId))
+            })
+            .expect((res) => {
+                assert.equal(res.body.success, true);
+            })
+            .expect(200);
+    });
+
+});
+
+describe('after deploying contract', () => {
+    let accountId = 'testing' + Date.now();
+    let method = twoFactorMethods[0];
+    let securityCode = '';
+    let requestId = -1;
+    let testContractDeployed = true;
+
+    test('initCode for an account, sets up 2fa method', async () => {
+        await createNearAccount(accountId);
+
+        await request.post('/2fa/init')
+            .send({
+                accountId,
+                method,
+                ...(await signatureFor(accountId))
+            })
+            .expect('Content-Type', /json/)
+            .expect((res) => {
+                assert.equal(res.body.success, true);
+                securityCode = getCodeFromLogs();
+            })
+            .expect(200);
+    });
+
+    test('initCode for an account with contract deployed', async () => {
+
+        await request.post('/2fa/init')
+            .send({
+                testContractDeployed,
+                accountId,
+                method,
+                ...(await signatureFor(accountId))
+            })
+            .expect(401);
+    });
+
+    test('verify 2fa method', async () => {
+
+        await request.post('/2fa/verify')
+            .send({
+                accountId,
+                requestId,
+                securityCode,
+                ...(await signatureFor(accountId))
+            })
+            .expect('Content-Type', /json/)
+            .expect((res) => {
+                assert.equal(res.body.success, true);
+            })
+            .expect(200);
+    });
+
+});
+
+
+describe('code older than 5min should fail', () => {
+    let accountId = 'testing' + Date.now();
+    let method = twoFactorMethods[0];
+    let securityCode = '';
+    let requestId = -1;
+
+    test('initCode for an account, sets up 2fa method', async () => {
```

Comment:
should also test a case where security code is not sent at all (as it previously was allowing it –  easy to introduce regression)